citadel_workspace_service/src/kernel/mod.rs

```diff
@@ -171,6 +230,40 @@ impl NetKernel for CitadelWorkspaceService {
                     }
                 }
             }
+            NodeResult::ObjectTransferHandle(object_transfer_handle) => {
+                let object_transfer_handler = object_transfer_handle.handle;
+                let implicated_cid = object_transfer_handler.receiver;
+                let peer_cid = object_transfer_handler.source;
+                // TODO: Utilize metadata to store handler by object id
+                let object_id: u32 = object_transfer_handler.source as u32;
+
+                let mut server_connection_map = self.server_connection_map.lock().await;
+                if let Some(connection) = server_connection_map.get_mut(&implicated_cid) {
+                    connection.add_object_transfer_handler(
+                        peer_cid,
+                        object_id,
+                        Some(object_transfer_handler),
+                    );
+                    let uuid = connection.associated_tcp_connection;
+                    // TODO: add metadata to the request we send to TCP client
+                    let response =
+                        InternalServiceResponse::FileTransferRequest(FileTransferRequest {
+                            cid: implicated_cid,
+                            peer_cid,
+                            metadata: VirtualObjectMetadata {
+                                name: "".to_string(),
+                                date_created: "".to_string(),
+                                author: "".to_string(),
+                                plaintext_length: 0,
+                                group_count: 0,
+                                object_id,
+                                cid: 0,
+                                transfer_type: TransferType::FileTransfer,
```

Comment:
The transfer type may not be a FileTrasfer type, it may be an ReVFS, right? I think this field can be found inside the handle itself (see `is_revfs_pull`): https://avarok-cybersecurity.github.io/Citadel-Protocol/docs/prelude/struct.ObjectTransferHandler.html

`is_revfs_pull` is a bit misleading (I will need to make another PR for clarity in the future). This bool can be mapped:

`true`: `TransferType::REVFS`
`false`: `TransferType::FileTransfer`

```suggestion
                                transfer_type: handle.transfer_type
```

Reply:
Ah yes, it automatically generated the contents there. I was just generating Metadata to send as a placeholder for now